diff --git a/arch/arm64/boot/dts/renesas/r9a09g057h4-dev.dts b/arch/arm64/boot/dts/renesas/r9a09g057h4-dev.dts
index 67cc4a9..8524042 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g057h4-dev.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g057h4-dev.dts
@@ -190,6 +190,15 @@
 		regulator-max-microvolt = <1500000>;
 		regulator-always-on;
 	};
+
+	udmabuf@0 {
+		compatible = "ikwzm,u-dma-buf";
+		device-name = "udmabuf0";
+		minor-number = <0>;
+		size = <0x4000000>; // 64MiB
+		dma-coherent;
+		memory-region = <&image_buf0>;
+	};
 };
 
 &pinctrl {
diff --git a/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-alpha.dts b/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-alpha.dts
index 4473f24..868e72e 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-alpha.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-alpha.dts
@@ -333,6 +333,15 @@
 		};
 #endif
 	};
+
+	udmabuf@0 {
+		compatible = "ikwzm,u-dma-buf";
+		device-name = "udmabuf0";
+		minor-number = <0>;
+		size = <0x4000000>; // 64MiB
+		dma-coherent;
+		memory-region = <&image_buf0>;
+	};
 };
 
 &audio_clka {
diff --git a/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-ver1.dts b/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-ver1.dts
index af1731f..625f24e 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-ver1.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-ver1.dts
@@ -337,6 +337,15 @@
 		};
 #endif
 	};
+
+	udmabuf@0 {
+		compatible = "ikwzm,u-dma-buf";
+		device-name = "udmabuf0";
+		minor-number = <0>;
+		size = <0x4000000>; // 64MiB
+		dma-coherent;
+		memory-region = <&image_buf0>;
+	};
 };
 
 &audio_clka {
