diff --git a/arch/arm64/boot/dts/renesas/r9a09g057h4-dev.dts b/arch/arm64/boot/dts/renesas/r9a09g057h4-dev.dts
index 57925d5..0e89ef4 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g057h4-dev.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g057h4-dev.dts
@@ -184,6 +184,15 @@
 		regulator-max-microvolt = <1500000>;
 		regulator-always-on;
 	};
+
+	udmabuf@0 {
+		compatible = "ikwzm,u-dma-buf";
+		device-name = "udmabuf0";
+		minor-number = <0>;
+		size = <0x4000000>; // 64MiB
+		dma-coherent;
+		memory-region = <&image_buf0>;
+	};
 };
 
 &pinctrl {
diff --git a/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-alpha.dts b/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-alpha.dts
index 186fff2..b0c6159 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-alpha.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-alpha.dts
@@ -327,6 +327,15 @@
 		};
 #endif
 	};
+
+	udmabuf@0 {
+		compatible = "ikwzm,u-dma-buf";
+		device-name = "udmabuf0";
+		minor-number = <0>;
+		size = <0x4000000>; // 64MiB
+		dma-coherent;
+		memory-region = <&image_buf0>;
+	};
 };
 
 &audio_clka {
diff --git a/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-ver1.dts b/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-ver1.dts
index 46934db..54892e5 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-ver1.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-ver1.dts
@@ -309,6 +309,15 @@
 		};
 #endif
 	};
+
+	udmabuf@0 {
+		compatible = "ikwzm,u-dma-buf";
+		device-name = "udmabuf0";
+		minor-number = <0>;
+		size = <0x4000000>; // 64MiB
+		dma-coherent;
+		memory-region = <&image_buf0>;
+	};
 };
 
 &audio_clka {
